packages/playwright-test/src/reporters/azure.ts

```diff
@@ -0,0 +1,481 @@
+
+import type { TestPoint } from 'azure-devops-node-api/interfaces/TestInterfaces';
+import type { WebApi } from 'azure-devops-node-api';
+import { colors } from 'playwright-core/lib/utilsBundle';
+import { existsSync, readFileSync } from 'fs';
+import type { ICoreApi } from 'azure-devops-node-api/CoreApi';
+import type { TeamProject } from 'azure-devops-node-api/interfaces/CoreInterfaces';
+import { createGuid } from 'playwright-core/lib/utils';
+
+const Statuses = {
+  passed: 'Passed',
+  failed: 'Failed',
+  skipped: 'Paused',
+  timedOut: 'Failed'
+};
+
+const attachmentTypesArray = [
+  'screenshot',
+  'video',
+  'trace',
+] as const;
+
+type TAttachmentType = Array<typeof attachmentTypesArray[number]>;
+
+export interface AzureReporterOptions {
+  token: string;
+  planId: number;
+  orgUrl: string;
+  projectName: string;
+  logging?: boolean | undefined;
+  isDisabled?: boolean | undefined;
+  environment?: string | undefined;
+  testRunTitle?: string | undefined;
+  uploadAttachments?: boolean | undefined;
+  attachmentsType?: TAttachmentType | undefined;
+}
+
+interface TestResultsToTestRun {
+  statusCode: number;
+  result: Result;
+  headers: Headers;
+}
+interface Result {
+  count: number;
+  value?: ValueEntity[] | null;
+}
+interface ValueEntity {
+  id: number;
+  project: Project;
+  outcome: string;
+  testRun: TestRun;
+  priority: number;
+  url: string;
+  lastUpdatedBy: LastUpdatedBy;
+}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface Project {
+}
+interface TestRun {
+  id: string;
+}
+interface LastUpdatedBy {
+  displayName?: null;
+  id?: null;
+}
+interface Headers {
+  'cache-control': string;
+  'pragma': string;
+  'content-length': string;
+  'content-type': string;
+  'expires': string;
+  'p3p': string;
+  'x-tfs-processid': string;
+  'strict-transport-security': string;
+  'activityid': string;
+  'x-tfs-session': string;
+  'x-vss-e2eid': string;
+  'x-vss-senderdeploymentid': string;
+  'x-vss-userdata': string;
+  'x-frame-options': string;
+  'request-context': string;
+  'access-control-expose-headers': string;
+  'x-content-type-options': string;
+  'x-cache': string;
+  'x-msedge-ref': string;
+  'date': string;
+  'connection': string;
+}
+
+class AzureDevOpsReporter implements Reporter {
+  private testApi!: Test.ITestApi;
+  private coreApi!: ICoreApi;
+  private _options: AzureReporterOptions;
+  private publishedResultsCount = 0;
+  private resultsToBePublished: string[] = [];
+  private connection!: WebApi;
+  private orgUrl!: string;
+  private projectName!: string;
+  private environment?: string;
+  private planId = 0;
+  private runId!: number;
+  private logging = true;
+  private isDisabled = false;
+  private testRunTitle = '';
+  private uploadAttachments = false;
+  private attachmentsType?: TAttachmentType;
+  private token: string = '';
+
+  public constructor(options: AzureReporterOptions = {} as AzureReporterOptions) {
```

Comment:
```suggestion
  public constructor(options: Partial<AzureReporterOptions> = {}) {
```
I think thats what you actually want.